mlrun/package/packagers_manager.py

```diff
@@ -0,0 +1,632 @@
+# Copyright 2018 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import importlib
+import inspect
+import itertools
+import typing
+
+from mlrun.artifacts import Artifact
+from mlrun.datastore import DataItem, is_store_uri, parse_store_uri, store_manager
+from mlrun.errors import MLRunInvalidArgumentError
+from mlrun.package.packager import Packager
+from mlrun.utils import StorePrefix, logger
+
+
+class PackagersManager:
+    """
+    A packager manager is holding the project's packagers and sending them objects to pack and data items to unpack.
+
+    It prepares the instructions / log hint configurations and then looks for the first packager who fits the task.
+    That's why when the manager collects its packagers, it first collects builtin MLRun packagers and only then the
+    user's custom packagers, this way user's custom packagers will have higher priority.
+    """
+
+    # Mandatory packagers to be collected at initialization time:
+    _MLRUN_REQUIREMENTS_PACKAGERS = [
+        "python_standard_library",
+        "pandas",
+        "numpy",
+        "mlrun",
+    ]
+    # Optional packagers to be collected at initialization time:
+    _EXTENDED_PACKAGERS = ["matplotlib", "plotly", "bokeh"]
+
+    def __init__(self, default_packager: Packager = None):
+        """
+        Initialize a packagers manager.
+
+        :param default_packager: The default packager to use in case all packagers managed by the manager do not fit an
+                                 object or data item.
+        """
+        # Set the default packager:
+        self._default_packager = default_packager or Packager
+
+        # Set an artifacts list and results dictionary to collect all packed objects (will be used later to write extra
+        # data if noted by the user using the log hint key "extra_data")
+        self._artifacts: typing.List[Artifact] = []
+        self._results = {}
+
+        # Initialize the packagers list:
+        self._packagers: typing.List[Packager] = []
+
+        # Collect the builtin standard packagers:
+        self._collect_packagers()
```

Comment:
Suggest to log here the list of available packagers after the collection, will ease up debugging etc.

Reply:
will do